Debugging messages, refactor of APITest, and change of API.storeTransactions

dnck opened the pull request:
Mostly just log messages for trouble-shooting. 

But, also, 

* API.storeTransactionsStatement() sets the arrival time to millisecond precsion. For this reason, the APITest.java needed to be refactored.

dzhelezov commented on src/main/java/net/helix/pendulum/service/snapshot/impl/SnapshotProviderImpl.java:
```diff
@@ -209,6 +209,7 @@ private void loadSnapshots() throws SnapshotException, SpentAddressesException {
         }
 
         latestSnapshot = initialSnapshot.clone();
+        log.debug("Initial snapshot time: {}", latestSnapshot.getInitialTimestamp());
```
How often is this going to happen? Perhaps `trace` is better here?

dnck commented:
Changes made:

* Moved to trace for instrumentation logs
* Changed the logback-save.xml to send trace messages to log file by default 
* Changed the PendulumIOUtils.saveLogs method to allow for different logback xml files supplied on command line by user (e.g. -Dlogback.configurationFile pathToAlternative.xml)
